packages/client/lib/functions/generateRandomPlacements.ts

```diff
@@ -1,34 +1,42 @@
-import { initialData } from "../constants";
-import { playerData } from "../interfaces/types";
+import { initialData, shipLengths } from "../constants";
+import { playerData, ShipInfo, shipLength } from "../interfaces/types";
 import placeShip from "./placeShip";
 
 /**Places ships from randomly generated coordinates that is not outside the grid or already occupied on any provided 10*10 grid*/
 const generateRandomPlacements = (
-  board: number[][],
-  setPlayerData: React.Dispatch<React.SetStateAction<playerData>>
-): number[][] => {
+  setPlayerData: React.Dispatch<React.SetStateAction<playerData>>,
+  playerData: playerData
+) => {
+  let shipInfo: ShipInfo[] = JSON.parse(JSON.stringify(playerData.shipInfo));
+  let finalShipInfo: ShipInfo[];
+
   for (let shipIndex = 0; shipIndex < initialData.length; shipIndex++) {
     const width = 10;
-    let placed = false;
+    let placed: ShipInfo[] | undefined = undefined;
     while (!placed) {
       const orientation =
         Math.round(Math.random()) >= 0.5 ? "horizontal" : "vertical";
 
       const x = Math.floor(Math.random() * width);
       const y = Math.floor(Math.random() * width);
 
-      console.log(shipIndex);
+      console.log(
+        "Placing ship of length",
+        shipLengths[shipIndex],
+        orientation
+      );
       console.log("[" + x, y + "]");
 
-      if (
-        placeShip(board, shipIndex, x, y, orientation, setPlayerData) === false
-      ) {
-        placed = false;
-      } else placed = true;
+      placed = placeShip(shipIndex, [x, y], orientation, shipInfo);
+    }
+    if (placed) {
+      finalShipInfo = placed;
+    } else {
+      throw new Error("No ship placed");
     }
   }
 
-  return board;
+  setPlayerData((prev) => ({ ...prev, shipInfo: finalShipInfo }));
```

Comment:
I moved `setPlayerData` out of `placeShip` to here to call the function once instead of multiple times in `placeShip`, but I JUST realized that this breaks manual placing. Fix for me ty :D

Reply:
ah, okay. I can give it a go.